server/main.lua

```diff
@@ -78,98 +94,105 @@ end)
 end)
 
 QBCore.Functions.CreateCallback("qb-multicharacter:server:GetNumberOfCharacters", function(source, cb)
     local src = source
-    local license = QBCore.Functions.GetIdentifier(src, 'license')
-    local numOfChars = 0
-
-    if next(Config.PlayersNumberOfCharacters) then
-        for i, v in pairs(Config.PlayersNumberOfCharacters) do
-            if v.license == license then
-                numOfChars = v.numberOfChars
-                break
-            else 
-                numOfChars = Config.DefaultNumberOfCharacters
-            end
-        end
+    local validCoords = ValidateHidden(GetEntityCoords(GetPlayerPed(src)))
+    if validCoords then
+        local numOfChars = PlayerMaxCharacters(src)
+        cb(numOfChars)
     else
-        numOfChars = Config.DefaultNumberOfCharacters
+        TriggerEvent('qb-log:server:CreateLog', 'multicharacter', 'multicharacter', "red", "**"..GetPlayerName(src) .. " (id: "..src..")** has attempted to get number of characters while not in the starting interior location")
     end
-    cb(numOfChars)
 end)
 
 QBCore.Functions.CreateCallback("qb-multicharacter:server:setupCharacters", function(source, cb)
-    local license = QBCore.Functions.GetIdentifier(source, 'license')
-    local plyChars = {}
-    MySQL.Async.fetchAll('SELECT * FROM players WHERE license = ?', {license}, function(result)
-        for i = 1, (#result), 1 do
-            result[i].charinfo = json.decode(result[i].charinfo)
-            result[i].money = json.decode(result[i].money)
-            result[i].job = json.decode(result[i].job)
-            plyChars[#plyChars+1] = result[i]
-        end
-        cb(plyChars)
-    end)
+    local src = source
+    local validCoords = ValidateHidden(GetEntityCoords(GetPlayerPed(src)))
+    if validCoords then
+        local license = QBCore.Functions.GetIdentifier(src, 'license')
+        local limit = PlayerMaxCharacters(src)
+        local plyChars = {}
+        MySQL.Async.fetchAll('SELECT * FROM players WHERE license = ? LIMIT ?', {license, limit}, function(result)
+            for i = 1, (#result), 1 do
+                result[i].charinfo = json.decode(result[i].charinfo)
+                result[i].money = json.decode(result[i].money)
+                result[i].job = json.decode(result[i].job)
+                plyChars[#plyChars+1] = result[i]
+            end
+            cb(plyChars)
+        end)
+    else
+        TriggerEvent('qb-log:server:CreateLog', 'multicharacter', 'multicharacter', "red", "**"..GetPlayerName(src) .. " (id: "..src..")** has attempted to get list characters while not in the starting interior location")
+    end
 end)
 
 QBCore.Functions.CreateCallback("qb-multicharacter:server:getSkin", function(source, cb, cid)
-    local result = MySQL.Sync.fetchAll('SELECT * FROM playerskins WHERE citizenid = ? AND active = ?', {cid, 1})
+    local result = MySQL.Sync.fetchAll('SELECT * FROM playerskins WHERE citizenid = ? AND active = ? LIMIT 1', {cid, 1})
```

Comment:
You can use fetchSingle here to get one result instead of limiting it to 1

Reply:
easy as, got that in there now